coffeecard/CoffeeCard.Library/Services/TokenService.cs

```diff
@@ -44,30 +47,92 @@ public JwtSecurityToken ReadToken(string token)
+        /// Receives the serialized version of a JWT token as a string.
+        /// Checks if the JWT token is valid (Based on its lifetime)
         /// </summary>
         /// <param name="tokenString"></param>
-        /// <returns></returns>
-        public async Task<bool> ValidateToken(string tokenString)
+        /// <returns>bool</returns>
+        public bool ValidateToken(string tokenString)
         {
+            var securityKey = new SymmetricSecurityKey(
+                Encoding.UTF8.GetBytes(_identitySettings.TokenKey));
+
             try
             {
-                var tokenNotUsed = false;
+                var securityTokenHandler = new JwtSecurityTokenHandler();
+                var validationParameters = new TokenValidationParameters
+                {
+                    ValidateAudience = false,
+                    ValidateIssuer = false,
+                    ValidateIssuerSigningKey = true,
+                    IssuerSigningKey = securityKey,
+                    ValidateLifetime = true,
+                    ClockSkew = TimeSpan.Zero //the default for this setting is 5 minutes
+                };
+
+                securityTokenHandler.ValidateToken(tokenString, validationParameters, out _); // Throws exception if token is invalid
+            }
+            catch (Exception e) when (e is ArgumentException ||
+                                      e is SecurityTokenException)
+            {
+                Log.Information("Received invalid token");
+                return false;
+            }
+
+            return true;
+        }
+        
+        /// <summary>
+        /// Receives the serialized version of a JWT token as a string.
+        /// Checks if the JWT token is valid (Based on its lifetime and if it has been used before)
+        /// </summary>
+        /// <param name="tokenString"></param>
+        /// <returns>bool</returns>
+        public async Task<bool> ValidateTokenIsUnusedAsync(string tokenString)
+        {
+            try
+            {
+                var tokenIsUnused = false;
                 var token = ReadToken(tokenString);
 
                 var user = await _claimsUtilities.ValidateAndReturnUserFromEmailClaimAsync(token.Claims);
 
-                if (user.Tokens.Contains(new Token(tokenString))) tokenNotUsed = true;
+                if (user.Tokens.Contains(new Token(tokenString))) tokenIsUnused = true; // Tokens are removed from the user on account recovery
 
-                return token.ValidTo > DateTime.UtcNow && tokenNotUsed;
+                return ValidateToken(tokenString) && tokenIsUnused;
             }
             catch (ArgumentException e)
             {
-                Log.Error($"Unable to read token. Exception thrown = {e}");
+                Log.Error(e, "Unable to read token. Exception thrown");
                 return false;
             }
         }
+        
+        public string ValidateTokenAndGetEmail(string token)
```

Comment:
Consider renaming to `ValidateVerificationTokenAndReadEmail` since method only applies for verification tokens